modules/phase_field/src/materials/ComputePolycrystalElasticityTensor.C

```diff
@@ -0,0 +1,162 @@
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "ComputePolycrystalElasticityTensor.h"
+#include "AddV.h"
+#include "RotationTensor.h"
+#include "GrainTracker.h"
+
+template<>
+InputParameters validParams<ComputePolycrystalElasticityTensor>()
+{
+  InputParameters params = validParams<ComputeElasticityTensorBase>();
+  params.addClassDescription("Compute an evolving elasticity tensor coupled to a grain growth phase field model.");
+  params.addRequiredParam<FileName>("Euler_angles_file_name", "Name of the file containing the Euler angles");
+  params.addParam<Real>("length_scale", 1.0e-9, "Lengthscale of the problem, in meters");
+  params.addParam<Real>("pressure_scale", 1.0e6, "Pressure scale of the problem, in pa");
+  params.addRequiredParam<unsigned int>("op_num", "number of order parameters");
+  params.addRequiredParam<std::string>("var_name_base", "base for variable names");
+  params.addCoupledVar("v", "Array of coupled variables");
+  params.addRequiredParam<UserObjectName>("GrainTracker_object", "The GrainTracker UserObject to get values from.");
+  params.addRequiredParam<unsigned int>("grain_num", "Number of initial grains that will be modeled");
+  params.addParam<unsigned int>("stiffness_buffer",10,"Number of extra elastic stiffnesses that are created to handle new grains");
+  params.addRequiredParam<std::vector<Real> >("Elastic_constants", "Vector containing elastic constants for fill method");
+  params.addParam<MooseEnum>("fill_method", RankFourTensor::fillMethodEnum() = "symmetric9", "The fill method");
+  return params;
+}
+
+ComputePolycrystalElasticityTensor::ComputePolycrystalElasticityTensor(const std::string & name,
+                                                 InputParameters parameters) :
+    ComputeElasticityTensorBase(name, AddV(parameters) ),
+    _C_unrotated(getParam<std::vector<Real> >("Elastic_constants"), (RankFourTensor::FillMethod)(int)getParam<MooseEnum>("fill_method")),
+    _length_scale(getParam<Real>("length_scale")),
+    _pressure_scale(getParam<Real>("pressure_scale")),
+    _Euler_angles_file_name(getParam<FileName>("Euler_angles_file_name")),
+    _grain_tracker(getUserObject<GrainTracker>("GrainTracker_object")),
+    _grain_num(getParam<unsigned int>("grain_num")),
+    _stiffness_buffer(getParam<unsigned int>("stiffness_buffer")),
+    _JtoeV(6.24150974e18), // Joule to eV conversion
+    _nop(coupledComponents("v"))
+{
+  // Initialize values for crystals
+  _vals.resize(_nop);
+  _D_elastic_tensor.resize(_nop);
+  _C_rotated.resize(_grain_num + _stiffness_buffer);
+
+  // Read in Euler angles from "angle_file_name"
+  std::ifstream inFile(_Euler_angles_file_name.c_str());
+
+  if (!inFile)
+    mooseError("Can't open input file ");
+
+  for (unsigned int i = 0; i < 4; ++i)
+    inFile.ignore(255, '\n'); // ignore line
```

Comment:
The ```255``` should really be ```std::numeric_limits<std::streamsize>::max()``` as we do not want to but a limit on the number of characters skipped (if there were long comments for example)